### DISCUSSION TITLE
Inline system file editing (via fileupload widget)

### DISCUSSION THREAD

**Discussion by lex0r** [wintercms/winter#670] [Ideas]
Even though it may not be particularly useful by itself, inline editing while uploading files using fileupload widget opens new options like assigning additional attributes without doing the click to open the edit popup. This can be useful when you need to upload tens of files and want to quickly edit their attributes while seeing them all in the list, which is a way to avoid mistakes with large uploads (quick table-like overview of files and attributes without a need to open a popup).

**Comment by LukeTowers** [comment]
Any rough designs of how that would look / function visually?

**Reply by lex0r** [reply]
Visually, just literally same system file config form rendered inline:
![image](https://user-images.githubusercontent.com/966581/188896180-1ca8a950-8b70-4b0a-87dc-c213945d3e5d.png)

On the visual it's extended with extra things like tags and has some other minor modifications.

Once the form is submitted the files are updated with the information provided for each of them, i.e. we delay file record updates until the form is sent.

**Reply by lex0r** [reply]
@LukeTowers  @bennothommo anyone else thinking this could be useful in the core? I'm good either way ofc

**Reply by bennothommo** [reply]
Sorry, I missed this before. I like this idea, and although I still think having additional attributes for the files be part of the popup window might be more "clean", I certainly can see the benefits of making additional fields inline available to developers.

Feel free to send a PR :)

**Reply by LukeTowers** [reply]
I'm fine with it so long as it's an option that's disabled by default. Looks good!
